src/ngx_http_lua_pipe.c

```diff
@@ -688,7 +688,7 @@ ngx_http_lua_ffi_pipe_spawn(ngx_http_lua_ffi_pipe_proc_t *proc,
         /* close listening socket fd */
         ls = ngx_cycle->listening.elts;
         for (i = 0; i < ngx_cycle->listening.nelts; i++) {
-            if (ngx_close_socket(ls[i].fd) == -1) {
+            if (ls[i].fd != -1 && ngx_close_socket(ls[i].fd) == -1) {
```

Comment:
```suggestion
            if (ls[i].fd != (ngx_socket_t) -1 && ngx_close_socket(ls[i].fd) == -1) {
```

Maybe slightly clearer even if we should be fine here in UNIX.